api/client/request.go

```diff
@@ -302,17 +306,25 @@ func (r *Request) Do() *Response {
 	}
 }
 
-func handleServiceUnavailable(resp *http.Response, attemptNum int) {
+func handleServiceUnavailable(resp *http.Response, attemptNum int) bool {
```

Comment:
If it is returning a `bool` then the name of the function should change to a question, something like `retryHeaderReceived()` that way the `if` statement is easier to read.

Reply:
sure will address that.